### PR TITLE
add new unstable param check

### PR DESCRIPTION
**Pull request by JJSwigut**
Added a check for if an unstable (not just dynamic) parameter has been added to a composable where it didn't previously exist.

### CODE REVIEW

File: compose-guard/gradle-plugin/src/functionalTest/kotlin/com/joetr/compose/guard/task/ComposeCompilerReportCheckTaskTest.kt [hunk 1]
```diff
@@ -345,4 +345,58 @@ class ComposeCompilerReportCheckTaskTest {
...
+            package com.example.myapplication
+
+            import androidx.compose.material3.Text
+            import androidx.compose.runtime.Composable
+
+            data class UnstableClass(var value: String)
+            data class OtherUnstableClass(var value: String)
+
+            @Composable
+            fun TestComposable(firstUnstable: UnstableClass, newUnstable: OtherUnstableClass) {
```
Comment (j-roskopf) [inline compose-guard/gradle-plugin/src/functionalTest/kotlin/com/joetr/compose/guard/task/ComposeCompilerReportCheckTaskTest.kt]: can you either add another test or modify the existing one to delete an existing unstable class too and verifies it still triggers? we still want to make sure it triggers if not just the absolute amount of unstable classes increase

File: compose-guard/gradle-plugin/src/main/kotlin/com/joetr/compose/guard/ComposeChecks.kt [hunk 2]
```diff
@@ -139,4 +141,51 @@ public object ComposeChecks {
...
+
+        // find new unstable parameters by removing all golden parameters from checked parameters
+        checkedUnstableParamsMap.keys.removeAll(goldenUnstableParamsMap.keys)
+
+        if (checkedUnstableParamsMap.isNotEmpty()) {
+            throw GradleException(
+                "New unstable parameters were added! \n" +
+                    checkedUnstableParamsMap.keys.joinToString(separator = ",") + "\n" +
+                    "More info: https://github.com/androidx/androidx/blob/androidx-main/compose/compiler/design/compiler-metrics.md" +
+                    "#parameters-that-are-unstable",
```
Comment (j-roskopf) [inline compose-guard/gradle-plugin/src/main/kotlin/com/joetr/compose/guard/ComposeChecks.kt]: this url doesn't not appear to be valid :) all of the other ones are actual anchors on that page

**Review by j-roskopf — DISMISSED** [review]

**Comment by j-roskopf** [comment]
also want to modify the readme to move the TODO for this task to TODONE?

File: compose-guard/gradle-plugin/src/functionalTest/kotlin/com/joetr/compose/guard/task/ComposeCompilerReportCheckTaskTest.kt (hunk 1 above)
Comment (JJSwigut) [reply compose-guard/gradle-plugin/src/functionalTest/kotlin/com/joetr/compose/guard/task/ComposeCompilerReportCheckTaskTest.kt]: I updated the check and the test so now it's actually checking if any new unstable param was added, even if it's replacing the old unstable param. This was absolute amount really doesn't matter. I think that satisfies what you're talking about, right?

Comment (j-roskopf) [reply compose-guard/gradle-plugin/src/functionalTest/kotlin/com/joetr/compose/guard/task/ComposeCompilerReportCheckTaskTest.kt]: absolutely

**Pull request merged**